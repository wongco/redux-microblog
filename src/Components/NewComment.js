import React, { Component } from 'react';
//import styled from 'styled-components';

class NewComment extends Component {
  handleSubmit = evt => {
    evt.preventDefault();
    this.props.addComment(this.state.comment);
    this.setState({ comment: '' });
  };

  handleChange = evt => {
    // runs on every keystroke
    this.setState({
      [evt.target.name]: evt.target.value
    });
  };

  render() {
    return (
      <div className="NewComment">
        <form onSubmit={this.handleSubmit}>
          <input
            type="text"
            name="comment"
            value={this.state.comment}
            onChange={this.handleChange}
            placeholder="New Comment"
          />
          <button>Add</button>
        </form>
      </div>
    );
  }

  state = {
    comment: ''
  };
}

NewComment.propTypes = {};

NewComment.defaultProps = {
  addComment: () => console.log('Adding new comment')
};

export default NewComment;
