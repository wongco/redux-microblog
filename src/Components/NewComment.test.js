import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import NewComment from './NewComment';

it('renders without crashing', () => {
  shallow(<NewComment />);
});

it('matches snapshot for default criteria', () => {
  const wrapper = shallow(<NewComment />);
  const serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});
