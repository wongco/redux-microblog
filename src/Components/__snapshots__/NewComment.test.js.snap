// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for default criteria 1`] = `
<div
  className="NewComment"
>
  <form
    onSubmit={[Function]}
  >
    <input
      name="comment"
      onChange={[Function]}
      placeholder="New Comment"
      type="text"
      value=""
    />
    <button>
      Add
    </button>
  </form>
</div>
`;
